File: docs/demo/lobs/oci/fgetdir.c
/* This file is installed in the following path when you install */
/* the database: $ORACLE_HOME/rdbms/demo/lobs/oci/fgetdir.c */

/* Getting the directory alias and filename */
#include <oratypes.h>
#include "lobdemo.h"
void BfileGetDir_proc(OCILobLocator *Bfile_loc, OCIEnv *envhp,
                      OCIError *errhp, OCISvcCtx *svchp, OCIStmt *stmthp)
{ 
   OraText dir_alias[32];
   OraText filename[256];
   ub2 d_length = 32;
   ub2 f_length = 256;

   printf ("----------- OCILobFileGetName Demo --------------\n"); 
   checkerr(errhp, OCILobFileGetName(envhp, errhp, Bfile_loc,
                       dir_alias, &d_length, filename, &f_length));

   dir_alias[d_length] = '\0';
   filename[f_length] = '\0';
   printf("Directory Alias : [%s]\n", dir_alias);
   printf("File name : [%s]\n", filename);
}

File: docs/demo/lobs/java/fgetdir.java
/* This file is installed in the following path when you install */
/* the database: $ORACLE_HOME/rdbms/demo/lobs/java/fgetdir.java */

/* Getting the directory alias and filename of a BFILE 
 * Uses Oracle proprietary classes.
 * In JDK5 invoke with java -Djdbc.drivers=oracle.jdbc.OracleDriver fgetdir
*/

import java.sql.Connection;
import java.sql.Statement;
import java.sql.ResultSet;
import java.sql.SQLException;

import oracle.sql.BFILE;
import oracle.jdbc.OracleResultSet;

public class fgetdir
{
  public static void main (String args [])
    throws Exception
  {
    Connection conn = LobDemoConnectionFactory.getConnection();
    Statement stmt = conn.createStatement ();
    ResultSet rset = stmt.executeQuery (
       "SELECT ad_graphic FROM Print_media" 
       + " WHERE product_id = 3106 AND ad_id = 13001");
    if (rset.next())
    {
      BFILE bfile = ((OracleResultSet)rset).getBFILE (1);
      System.out.println("Directory alias: " + bfile.getDirAlias());
      System.out.println("File name: " + bfile.getName());
    }
    stmt.close();
    conn.close();
  }
}
